tests/nat/builder/test_per_user_builder.py

```
@@ -0,0 +1,1225 @@
+
+async def test_per_user_builder_function_groups_expose_functions():
+    """Test that per-user function groups can expose functions."""
+    from nat.builder.function import FunctionGroup
+    from nat.builder.function import LambdaFunction
+    from nat.cli.register_workflow import register_per_user_function_group
+
+    class TestExposeConfig(FunctionGroupBaseConfig, name="test_expose_fg"):
+        pass
+
+    @register_per_user_function_group(config_type=TestExposeConfig)
+    async def test_expose_fg(config: TestExposeConfig, builder: Builder):
```

Ruff (0.14.8)

Unused function argument: `builder`

(ARG001)